parser.ts

```diff
@@ -0,0 +1,1102 @@
+      var length = buffer.length;
+      var a = (length < 128) ? ByteStream.getArray(length) : new Array(length);
+      var i = 0, j = 0;
+      var o = 0;
+      var e = o + length;
+      var buffer = buffer;
+      while (o < e) {
+        var x = buffer[o++];
+        if (x <= 0x7f) {
+          // Code points in the range '\u0001' to '\u007F' are represented by a
+          // single byte.
+          // The 7 bits of data in the byte give the value of the code point
+          // represented.
+          a[j++] = String.fromCharCode(x);
+        } else if (x <= 0xdf) {
+          // The null code point ('\u0000') and code points in the range '\u0080'
+          // to '\u07FF' are represented by a pair of bytes x and y.
+          var y = buffer[o++]
+          a[j++] = String.fromCharCode(((x & 0x1f) << 6) + (y & 0x3f));
+        } else {
+          // Code points in the range '\u0800' to '\uFFFF' are represented by 3
+          // bytes x, y, and z.
+          var y = buffer[o++];
+          var z = buffer[o++];
+          a[j++] = String.fromCharCode(((x & 0xf) << 12) + ((y & 0x3f) << 6) + (z & 0x3f));
+        }
+      }
+      if (j !== a.length) {
+        var b = (j < 128) ? ByteStream.getArray(j) : new Array(j);
+        for (var i = 0; i < j; i++) {
+          b[i] = a[i];
+        }
+        a = b;
+      }
+      return a.join("");
+    }
+
+    static readString(buffer: Uint8Array) {
+      var length = buffer.length;
+      if (length === 1) {
+        var c = buffer[0];
+        if (c <= 0x7f) {
+          return String.fromCharCode(c);
+        }
+      } else if (length < 128) {
+        return ByteStream.readStringFast(buffer);
+      }
+      return ByteStream.readStringSlow(buffer);
+    }
+
+    static readStringSlow(buffer: Uint8Array) {
+      // First try w/ TextDecoder, fallback to manually parsing if there was an
+      // error. This will handle parsing errors resulting from Java's modified
+      // UTF-8 implementation.
+      try {
+        return util.decodeUtf8Array(buffer);
+      } catch (e) {
+        return this.readStringFast(buffer);
+      }
+    }
+
+    static readU16(buffer: Uint8Array, o: number): number {
+      return buffer[o] << 8 | buffer[o + 1];
+    }
+  }
+
+  export enum ACCESS_FLAGS {
+    ACC_PUBLIC = 0x0001,
+    ACC_PRIVATE = 0x0002,
+    ACC_PROTECTED = 0x0004,
+    ACC_STATIC = 0x0008,
+    ACC_FINAL = 0x0010,
+    ACC_SYNCHRONIZED = 0x0020,
+    ACC_VOLATILE = 0x0040,
+    ACC_TRANSIENT = 0x0080,
+    ACC_NATIVE = 0x0100,
+    ACC_INTERFACE = 0x0200,
+    ACC_ABSTRACT = 0x0400
+  }
+
+  export enum TAGS {
+    CONSTANT_Class = 7,
+    CONSTANT_Fieldref = 9,
+    CONSTANT_Methodref = 10,
+    CONSTANT_InterfaceMethodref = 11,
+    CONSTANT_String = 8,
+    CONSTANT_Integer = 3,
+    CONSTANT_Float = 4,
+    CONSTANT_Long = 5,
+    CONSTANT_Double = 6,
+    CONSTANT_NameAndType = 12,
+    CONSTANT_Utf8 = 1,
+    CONSTANT_Unicode = 2,
+    CONSTANT_Any = 13 // NON-STANDARD
+  }
```

Comment:
The enum declarations look just like they appear in the JVM spec, I don't see how this can be attributed to any one project.

What other code are you referring to in parser.ts? We pretty much rewrote the entire file from scratch.